Добавление тестов для сайта Stellar Burgers

Lizabudka opened the pull request:
Добавлены файлы с тестами и файл с фикстурами
Добавлен .gitignore

lizasoma committed:
Добавление тестов для сайта

irisqul commented on tests/conftest.py:
```diff
@@ -0,0 +1,60 @@
+
+@pytest.fixture(scope='function')
+def login_data():
+    login_data = {'login': 'lizafrolova11234@mail.ru',
```
Нужно исправить: нам не нужны фикстуры чтобы получить статичные данные, нужно перенести в константы и просто получать по имени, а не вызывать функции. В фикстурах можно, например, генерировать новые данные для тестирования, уникальные для каждого теста.

irisqul commented on tests/conftest.py:
```diff
@@ -0,0 +1,60 @@
+    password = random_password
+
+    driver = webdriver.Chrome()
+    driver.get('https://stellarburgers.nomoreparties.site/register')
```
Нужно исправить: вынести урлы в константы и переиспользовать их

irisqul commented on tests/conftest.py:
```diff
@@ -0,0 +1,60 @@
+@pytest.fixture(scope='function')
+def log_in(login_data):
+    driver = webdriver.Chrome()
+    driver.get('https://stellarburgers.nomoreparties.site/login')
```
Нужно исправить: выделить в отдельную фикстуру запуск и завершение работы драйвера, и не дублировать этот код в других фикстурах

irisqul commented on tests/test_click_on_sections.py:
```diff
@@ -0,0 +1,51 @@
+
+
+def test_click_on_section_sauce_jumping_to_section_sauce():
+    driver = webdriver.Chrome()
```
Нужно исправить: запуск, завершение браузера и возможно открытие страницы - нужно перенести в фикстуру

irisqul commented on tests/conftest.py:
```diff
@@ -0,0 +1,60 @@
+    driver.get('https://stellarburgers.nomoreparties.site/register')
+
+    WebDriverWait(driver, 3).until(expected_conditions.visibility_of_element_located((
+        By.XPATH, './/*[text() = "Имя"]/parent::div/input')))  # Поле ввода 'Имя'
```
Нужно исправить: согласно заданию нужно использовать локаторы - Определи и опиши локаторы, которые будешь использовать в тестах. Создай файл, в котором сохранишь описание элементов. Добавь к каждому локатору комментарий, чтобы было проще ориентироваться в них. А в тестах нужно просто указывать имя локатора, а не каждый раз путь до него

lizasoma committed:
1\ Все локаторы и ссылки перенесены в отдельный файл и записаны в переменные
2\ Добавлена отдельная фикстура открытия и закрытия браузера
3\ Статичные логин и парлоль перенесены в переменные
